lib/file-source/src/file_server.rs

```diff
@@ -139,12 +138,11 @@ where
         let emitter = self.emitter.clone();
         let checkpointer = Arc::new(checkpointer);
         let sleep_duration = self.glob_minimum_cooldown;
-        self.handle.spawn(async move {
-            let mut done = false;
+        let checkpoint_task_handle = self.handle.spawn(async move {
             loop {
                 let sleep = sleep(sleep_duration);
                 tokio::select! {
-                    _ = &mut shutdown2 => done = true,
+                    _ = &mut shutdown2 => return checkpointer,
```

Comment:
`checkpointer` is an `Arc` box at this point. Is passing it in and returning it better than the surrounding code just adding another reference?

Reply:
Not necessarily, but it would be nice if we enforced a single writer for the checkpoint file and this is more compatible with that design should we choose to refactor.